lib/plug/router.ex

```diff
@@ -213,7 +213,8 @@ defmodule Plug.Router do
 
       @doc false
       def dispatch(%Plug.Conn{assigns: assigns} = conn, _opts) do
-        Map.get(conn.private, :plug_route).(conn)
+        {_path, fun} = Map.get(conn.private, :plug_route)
```

Comment:
Since `:plug_route` should be here, I think `Map.get/2` is not the optimal choice. If it returns `nil`, you'll have a nasty match error (which is fine since we expect the field there), but we don't communicate this requirement in this code. I'd go with either

```elixir
{_path, fun} = Map.fetch!(conn.private, :plug_route)
```

or with

```elixir
%{plug_route: {_path, fun}} = conn.private
```

(leaning on the first for my personal tastes)